### PR TITLE
Add community terms of service support

### PR DESCRIPTION
**Pull request by gdjohnson**
This pull adds community support for a TOS URI, which community admins can set in the community management modal.

This URI is then displayed whenever a user attempts to join a community with a new address, or accepts an email-based community invite.

It does not provide support for silent address-based invitations; see issue #1267. 

To test:
1. From an admin account, add a TOS link (should be valid http/s or will fail) to the community metadata form, in the "Manage Community" modal.
2. Add a new address to the community via the "Manage addresses"/SelectNewAddress modal
3. From an admin account, invite an email account (e.g. via mail.tm) to the community, then accept this invite by adding/connecting a new address to the community.

At each stage in which a user is connecting a new address to join the community, a link to the TOS should render. These steps should replicate on both on- and off-chain, and both public and private, communities.

### CODE REVIEW

File: client/scripts/views/components/header/login_selector.ts [hunk 1]
```diff
@@ -290,6 +290,8 @@ const LoginSelector: m.Component<{
       }
     }
 
+    const hasTermsOfService = app.chain ? app.chain.meta?.chain : app.community?.meta;
```
Comment (zakhap) [inline client/scripts/views/components/header/login_selector.ts:293]: This doesn't seem to be checking for the terms

Comment (zakhap) [reply client/scripts/views/components/header/login_selector.ts:293]: `const hasTermsOfService = !!activeInvite.terms;` like in the `confirm_invite_modal` seems to be what we want here

File: client/scripts/views/modals/select_address_modal.ts [hunk 2]
```diff
@@ -122,6 +121,12 @@ const SelectAddressModal: m.Component<{}, { selectedIndex: number, loading: bool
             ]),
           ])),
         ]),
+        !!meta.terms
```
Comment (zakhap) [inline client/scripts/views/modals/select_address_modal.ts:124]: we should standardize this check across the files touched in this PR. It should just be consistent

**Review by zakhap — CHANGES_REQUESTED** [review]
For the most part, this looks good. Just a few things to clean up before we can merge

File: client/scripts/views/modals/select_address_modal.ts (hunk 2 above)
Comment (gdjohnson) [reply client/scripts/views/modals/select_address_modal.ts:124]: SG! On it

File: client/scripts/views/components/header/login_selector.ts (hunk 1 above)
Comment (gdjohnson) [reply client/scripts/views/components/header/login_selector.ts:293]: Haha, yes, good catch

**Pull request merged**